api/handlers/exporter_test.go

```diff
@@ -465,3 +471,687 @@ func TestExporterDecideds_ErrorGetParticipantsInRange(t *testing.T) {
+			},
+			setupMock: func(store *mockTraceStore) {
+				pkBytes := common.Hex2Bytes("b24454393691331ee6eba4ffa2dbb2600b9859f908c3e648b6c6de9e1dea3e9329866015d08355c8d451427762b913d1")
+				var pk spectypes.ValidatorPK
+				copy(pk[:], pkBytes)
+				store.AddCommitteeDecided(phase0.Slot(150), pk, []uint64{1, 2, 3})
+			},
+			expectedStatus: http.StatusOK,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Data []*ParticipantResponse `json:"data"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				require.Len(t, resp.Data, 1)
+				assert.Equal(t, "ATTESTER", resp.Data[0].Role)
+				assert.Equal(t, uint64(150), resp.Data[0].Slot)
+				assert.Equal(t, []uint64{1, 2, 3}, resp.Data[0].Message.Signers)
+			},
+		}, {
+			name: "valid request - empty signers array - proposer role",
+			request: map[string]any{
+				"from":    100,
+				"to":      200,
+				"roles":   []string{"PROPOSER"},
+				"pubkeys": []string{"b24454393691331ee6eba4ffa2dbb2600b9859f908c3e648b6c6de9e1dea3e9329866015d08355c8d451427762b913d1"},
+			},
+			setupMock: func(store *mockTraceStore) {
+				pkBytes := common.Hex2Bytes("b24454393691331ee6eba4ffa2dbb2600b9859f908c3e648b6c6de9e1dea3e9329866015d08355c8d451427762b913d1")
+				var pk spectypes.ValidatorPK
+				copy(pk[:], pkBytes)
+
+				entry := qbftstorage.ParticipantsRangeEntry{
+					Signers: []uint64{},
+				}
+				store.AddValidatorDecided(spectypes.BNRoleProposer, phase0.Slot(150), pk, entry.Signers)
+			},
+			expectedStatus: http.StatusOK,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Data []*ParticipantResponse `json:"data"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				require.Len(t, resp.Data, 0) // Empty signers array should be filtered out
+			},
+		},
+		{
+			name: "valid request - empty signers array - attester role",
+			request: map[string]any{
+				"from":    100,
+				"to":      200,
+				"roles":   []string{"ATTESTER"},
+				"pubkeys": []string{"b24454393691331ee6eba4ffa2dbb2600b9859f908c3e648b6c6de9e1dea3e9329866015d08355c8d451427762b913d1"},
+			},
+			setupMock: func(store *mockTraceStore) {
+				pkBytes := common.Hex2Bytes("b24454393691331ee6eba4ffa2dbb2600b9859f908c3e648b6c6de9e1dea3e9329866015d08355c8d451427762b913d1")
+				var pk spectypes.ValidatorPK
+				copy(pk[:], pkBytes)
+
+				entry := qbftstorage.ParticipantsRangeEntry{
+					Signers: []uint64{},
+				}
+				store.AddCommitteeDecided(phase0.Slot(150), pk, entry.Signers)
+			},
+			expectedStatus: http.StatusOK,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Data []*ParticipantResponse `json:"data"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				require.Len(t, resp.Data, 0) // Empty signers array should be filtered out
+			},
+		},
+		{
+			name: "invalid request - invalid pubkey length",
```

Comment:
there are some other scenarios that can return BadRequest, like `request.From > request.To` and `len(request.Roles) == 0`  so it would be good to add tests for these as well